### PR TITLE
Added jekyll_environment to templating system

### PR DESCRIPTION
**Pull request by yasthil**
Allows us to set JEKYLL_ENV to production so our Google Analytics script can run

### CODE REVIEW

File: example-config.yml [hunk 1]
```diff
@@ -15,3 +15,4 @@ apps:
     hostname: www.local.bcc3d.ca
     google_analytics_key: UA-162929587-1
     google_maps_api_key:
+    jekyll_environment: #production
```
Comment (wsonnex) [inline example-config.yml:18]: A nicer pattern here is to use `production: true`, then use a ternary to insert the value `"production"`

The (very minor) problem with the current approach is that it involves a special value "production"

File: apps/bcc3d.ca/build.docker-compose.yml [hunk 2]
```diff
@@ -7,3 +7,5 @@ services:
       context: .
     volumes:
       - .:/build
+    environment:
```
Comment (wsonnex) [inline apps/bcc3d.ca/build.docker-compose.yml:10]: Please delete this file from the repo, and add it to .gitignore

File: example-config.yml (hunk 1 above)
Comment (yasthil) [reply example-config.yml]: Agreed, that's a much better pattern! I've made the necessary changes

File: example-config.yml [hunk 3]
```diff
@@ -15,4 +15,4 @@ apps:
     hostname: www.local.bcc3d.ca
     google_analytics_key: UA-162929587-1
     google_maps_api_key:
-    jekyll_environment: #production
\ No newline at end of file
+    production: true
```
Comment (wsonnex) [inline example-config.yml:18]: default was "false" before, which I think is correct, could you change it here?
ready for merge and deploy after this

Comment (yasthil) [reply example-config.yml:18]: Definitely! Sorry about that, will do the change now.

**Pull request merged**